ios/RNRadar.m

```diff
@@ -88,79 +111,539 @@ - (void)didFailWithStatus:(RadarStatus)status {
+    int radius;
+    if (radiusNumber != nil && [radiusNumber isKindOfClass:[NSNumber class]]) {
+        radius = [radiusNumber intValue];
+    } else {
+        radius = 1000;
+    }
+    NSArray *tags = optionsDict[@"tags"];
+    NSNumber *limitNumber = optionsDict[@"limit"];
+    int limit;
+    if (limitNumber != nil && [limitNumber isKindOfClass:[NSNumber class]]) {
+        limit = [limitNumber intValue];
+    } else {
+        limit = 10;
+    }
+
+    __block RCTPromiseResolveBlock resolver = resolve;
+    __block RCTPromiseRejectBlock rejecter = reject;
+
+    RadarSearchPointsCompletionHandler completionHandler = ^(RadarStatus status, CLLocation * _Nullable location, NSArray<RadarPoint *> * _Nullable points) {
+        if (status == RadarStatusSuccess && resolver) {
+            NSMutableDictionary *dict = [NSMutableDictionary new];
+            [dict setObject:[Radar stringForStatus:status] forKey:@"status"];
+            if (location) {
+                [dict setObject:[Radar dictionaryForLocation:location] forKey:@"location"];
+            }
+            if (points) {
+                [dict setObject:[RadarPoint arrayForPoints:points] forKey:@"points"];
+            }
+            resolver(dict);
+        } else if (rejecter) {
+            rejecter([Radar stringForStatus:status], [Radar stringForStatus:status], nil);
+        }
+        resolver = nil;
+        rejecter = nil;
+    };
+
+    if (near) {
+        [Radar searchPointsNear:near radius:radius tags:tags limit:limit completionHandler:completionHandler];
+    } else {
+        [Radar searchPointsWithRadius:radius tags:tags limit:limit completionHandler:completionHandler];
+    }
+}
+
+RCT_EXPORT_METHOD(autocomplete:(NSDictionary *)optionsDict resolve:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject) {
+    if (optionsDict == nil) {
+        if (reject) {
+            reject([Radar stringForStatus:RadarStatusErrorBadRequest], [Radar stringForStatus:RadarStatusErrorBadRequest], nil);
+        }
+
+        return;
+    }
+
+    NSDictionary *nearDict = optionsDict[@"near"];
+    if (nearDict == nil || ![nearDict isKindOfClass:[NSDictionary class]]) {
+        if (reject) {
+            reject([Radar stringForStatus:RadarStatusErrorBadRequest], [Radar stringForStatus:RadarStatusErrorBadRequest], nil);
+        }
+
+        return;
+    }
+
+    NSString *query = optionsDict[@"query"];
+    double latitude = [RCTConvert double:nearDict[@"latitude"]];
+    double longitude = [RCTConvert double:nearDict[@"longitude"]];
+    NSDate *timestamp = [NSDate new];
+    CLLocation *near = [[CLLocation alloc] initWithCoordinate:CLLocationCoordinate2DMake(latitude, longitude) altitude:-1 horizontalAccuracy:5 verticalAccuracy:-1 timestamp:timestamp];
+    NSNumber *limitNumber = optionsDict[@"limit"];
+    int limit;
+    if (limitNumber != nil && [limitNumber isKindOfClass:[NSNumber class]]) {
+        limit = [limitNumber intValue];
+    } else {
+        limit = 10;
```

Comment:
we shim 100 as the limit on the server if none is provided

Reply:
Seems high. We may want to change default to 10. Will Clubhouse